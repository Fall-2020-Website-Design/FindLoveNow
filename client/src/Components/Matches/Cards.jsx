import React, { Component } from 'react';
import Bar from './Bar';
import Arrows from './Arrows';
import Images from './Images';
import Row from 'react-bootstrap/Row';
import Col from 'react-bootstrap/Col';
import Card from 'react-bootstrap/Card';

import Image from 'react-bootstrap/Image';
import location_image from '../../Images/location.svg';

export class Cards extends Component {

    constructor(props) {
        super(props);
        this.state = { 
            currentPicture: this.props.user.currentPicture,
            name: this.props.user.name,
            age: this.props.user.age,
            bio: this.props.user.bio,
            miles: this.props.user.miles,
            images: this.props.user.images,
            width: 0
        }     
    }
    
    handleToUpdate = (index) => {
        this.setState({
            currentPicture: index
        });
    }

    nextImage = () => {
        this.setState({
            currentPicture: this.state.currentPicture === this.state.images.length-1 ? 0 : this.state.currentPicture + 1
        });
    }

    prevImage = () => {
        this.setState({
            currentPicture: this.state.currentPicture === 0 ? this.state.images.length-1 : this.state.currentPicture - 1
        });
    }

    render() {
        return (
            <div>
                <Col className="d-inline-flex justify-content-center pt-4">
                    <Card className="matches-card">
                        <Card.Body className="card-body">
                            <Bar total={this.state.images.length} current={this.state.currentPicture} click={this.handleToUpdate} />

                            <Arrows prev={this.prevImage} next={this.nextImage}/>

                            <Images image={this.state.images[this.state.currentPicture]} className="d-inline-flex justify-content-center " />

                            <Col className="info" >
                                <div className="name-age"> {this.state.name}, {this.state.age} </div>
                                <div className="bio"> {this.state.bio} </div> 
                            </Col>  

                            <div className="location">
                                <Row>
                                    <Image alt="" src={location_image} /> {this.state.miles} miles away
                                </Row>
                            </div>
                        </Card.Body>
                    </Card>
                </Col>
            </div>
        )
    }
}

export default Cards
