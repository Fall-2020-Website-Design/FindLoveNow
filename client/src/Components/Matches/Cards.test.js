import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { Cards } from './Cards';


Enzyme.configure({ adapter: new Adapter() });

describe('Cards', () => {
	let wrapper;

	beforeEach(() => {
		wrapper = shallow(<Cards />);
    });
    
    test('renders', () => {
		const component = renderer.create(
			<MemoryRouter>
				<Cards  />
			</MemoryRouter>
		).toJSON();
		
		expect(component).toMatchSnapshot();
    });
    

});