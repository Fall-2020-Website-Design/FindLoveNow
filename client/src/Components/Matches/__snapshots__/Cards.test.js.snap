// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cards renders 1`] = `
<div>
  <div
    className="d-inline-flex justify-content-center pt-4 col"
  >
    <div
      className="matches-card card"
    >
      <div
        className="card-body card-body"
      >
        <div>
          <div
            className="justify-content-center mt-1 row"
          />
        </div>
        <div>
          <div
            className="row"
          >
            <div
              className="d-flex justify-content-between arrows col"
            >
              <img
                alt="left"
                className="align-top left-arrow img-fluid"
                id="left"
                onClick={[Function]}
                src="arrow.svg"
              />
              <img
                alt="right"
                className="align-top right-arrow img-fluid"
                id="right"
                onClick={[Function]}
                src="arrow.svg"
              />
            </div>
          </div>
        </div>
        <div>
          <div
            className="justify-content-md-center row"
          >
            <div
              className="col"
            >
              <img
                className="card-img"
                src={null}
                style={
                  Object {
                    "height": "37rem",
                  }
                }
              />
            </div>
          </div>
        </div>
        <div
          className="info col"
        >
          <div
            className="name-age row"
          >
             
            name
            , 
            age
             - 
            0'0
             
          </div>
          <div
            className="edu row"
          >
             Education: 
            N\\A
             
          </div>
          <div
            className="bio row"
          >
             Phrase: 
            phrase
             
          </div>
        </div>
        <div
          className="location"
        >
          <div
            className="row"
          >
            <img
              alt=""
              className=""
              src="location.svg"
            />
             Location: 
            location
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
